src/framework/components/script/component.js

```diff
@@ -391,7 +391,7 @@ class ScriptComponent extends Component {
             // disable script if it fails to call method
             script.enabled = false;
 
-            if (!script._callbacks || !script._callbacks.error) {
+            if (!script?.hasEvent('error')) {
```

Comment:
Do you need the `?` here? This line above seems to assume `script` is valid.

Reply:
Good catch, I agree, it seems like an old edge case where it was possible that `EventHandler` deleted `_callbacks` :thinking: Or maybe it was always too defensive.